composer/loggers/mlflow_logger.py

```diff
@@ -151,26 +149,24 @@ def log_table(self, columns: List[str], rows: List[List[Any]], name: str = 'Tabl
             )
 
     def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
+        from mlflow import log_metrics
```

Comment:
fluent API uses the current active MLflow session, for Mosaic use cases, IIUC you create a run and record the `run_id`, then explicitly start with `run_id`, so we should be good here.

Reply:
ok sounds fine to me. lemme check in with @dakinggg

Reply:
IIRC @dbczumar or @prithvikannan added the optimized client and it was important for not slowing down runs due to mlflow logging. Why are we removing it?

Reply:
@dakinggg For the context we added async logging to MLflow earlier, and in this PR we are switching to using the async logging instead of the queue-based client, which is easier to read. And in the future, MLflow will switch to async logging by default.

Reply:
Ah ok, thanks for the context!

Reply:
yea @dakinggg we've basically added the custom async-logging logic we wrote in composer into MLflow directly, so this PR starts using that instead.